octomap_server/include/octomap_server/OctomapServer.h

```diff
@@ -88,7 +88,7 @@ class OctomapServer {
   typedef octomap_msgs::GetOctomap OctomapSrv;
   typedef octomap_msgs::BoundingBoxQuery BBXSrv;
 
-  OctomapServer(ros::NodeHandle private_nh_ = ros::NodeHandle("~"));
+  OctomapServer(const ros::NodeHandle &nh_ = ros::NodeHandle(), const ros::NodeHandle &private_nh_ = ros::NodeHandle("~"));
```

Comment:
Could you exchange the order or parameters? Otherwise this is a breaking API change that shouldn't get released into an active distro. What's worse, it might go even undetected in other code, where suddenly the Constructor would be called with a private NodeHandle as public one.